Sources/Sentry/SentryClient.m

```diff
@@ -108,32 +108,36 @@ - (NSString *_Nullable)captureMessage:(NSString *)message withScope:(SentryScope
 {
     SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelInfo];
     event.message = message;
-    if ([self.options.attachStacktrace boolValue]) {
-        [self attachStacktrace:event];
-    }
-    return [self captureEvent:event withScope:scope];
+    return [self sendEvent:event withScope:scope alwaysAttachStacktrace:NO];
```

Comment:
H: why are we overwriting the attach stack trace values? this is an options configuration
https://docs.sentry.io/error-reporting/configuration/?platform=csharp#attach-stacktrace

Reply:
We are not overriding them. For `captureError` and `captureException` we always attach it, no matter what is configured in the options. For `captureEvent` and `captureMessage` we take a look into the options. That's why I named it `alwaysAttachStacktrace`, but if it is not clear enough for you, please let me know how to make it clearer for you.

Reply:
@philipphofmann its not about the naming, actually this is about the options itself.
I don't recall the implementation on all SDKs, but on Java/Android it is like this:
the `attachStacktrace` options only play a role on stack traces for threads.
so even if you call `captureEvent` and your event has a custom exception, stack traces for the exception will be there, but stack traces for all the other threads will depend on `attachStacktrace`, so the implementation here looks different.
Not telling its wrong or anything, maybe it does make sense to be like this on Cocoa. so here is a question:

On android you can do captureEvent(SentryEvent(exception)) which should have a stack trace and not depend on `attachStacktrace`
is this possible in cocoa? if so, `captureEvent` should not be gated with `alwaysAttachStacktrace`

Reply:
On Cocoa we only attach the stacktrace for the current thread, because doing it for all threads is a bit complicated with SentryCrash. I changed it now, that when you capture an event with an exception the stacktrace is attached.